Add Alouette project, as a replacement for 12days

droberts-sea opened the pull request:
## Description
The 12 days of xmas weekend warrior didn't go so well for C6, so I've come up with a replacement.

### Differences
- Replace a christmas carol with a french-canadian children's song, to promote cultural diversity
- Get rid of some of the weird edge cases
- Add more scaffolding, both in the README and by providing some skeleton code
- Make tests more comprehensive / higher granularity
- Use spec-style testing instead of asserts

### Similarities
The theme is still building a cumulative song, so students still get to practice
- Loops within loops
- Functional decomposition
- String processing

## Still TODO
N/A

## Feedback Requested
- Does this address the key problems with 12 days?
- Are the instructions clear / tests correct / etc?

Commit:
Add README and lyrics for alouette

Commit:
Write some tests for Alouette

Commit:
Add skeleton code for Alouette

kariabancroft commented:
Do you think it would make sense to move this one to # 2 and delete 12 days completely? It's difficult but I think it should still be removed due to content.

droberts-sea commented:
Sounds good. Worst comes to worst and we need it back, it'll be in the git history.

kariabancroft commented on 08-alouette/alouette_test.rb:
```diff
@@ -0,0 +1,117 @@
+      end
+    end
+
+    it "generates the correct line for the first verse" do
```
By default, the other weekend warriors have all tests skipped except for the first one

Commit:
Remove the 12days project completely, replacing it with Alouette

Commit:
Merge remote-tracking branch 'adagold/master' into alouette

kariabancroft commented on 02-alouette/alouette.rb:
```diff
@@ -0,0 +1,21 @@
+class Alouette
```
I also don't think that we normally provide the rb file and the class and methods. Do we want to adjust them all or remove this one?

droberts-sea replied:
The main benefit to providing a class is to guide them into storing the song components in an array. I'm not sure how important that bit of scaffolding will be.

If we wanted still to provide it but to make the format consistent with the other weekend warrior projects, we could throw the array definition in the README, otherwise I'll just delete the file.

kariabancroft commented on 02-alouette/alouette_test.rb:
```diff
@@ -0,0 +1,117 @@
+    it "returns an array of strings" do
+      lines = Alouette.lines_for_verse(3)
+
+      lines.must_be_kind_of Array
```
The style of these tests is not the same as the other weekend warrior tests. I think we would only want to use "must" if we are planning on changing all of them from "assert"

kariabancroft commented on 02-alouette/alouette_test.rb:
```diff
@@ -0,0 +1,117 @@
+    end
+
+    it "generates the correct line for the first verse" do
+      Alouette.lines_for_verse(0).must_equal ['Et la tête!']
```
I think my other comment was removed when you updated the files but in other weekend warriors all tests but the first are skippped

Commit:
Skip all but the first test

Commit:
Convert from spec to assert style testing. First pass, no guarantee it works yet

Commit:
Get assert-style tests working

Clean up test names
Convert let to a @@class variable

Has been tested against my solution.

Commit:
Remove skeleton code, and add a request to think about the program's structure to the README
